x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/phases/frozen_phase.tsx

```diff
@@ -0,0 +1,210 @@
+
+interface Props {
+  setPhaseData: (key: keyof FrozenPhaseInterface & string, value: string | boolean) => void;
+  phaseData: FrozenPhaseInterface;
+  isShowingErrors: boolean;
+  errors?: PhaseValidationErrors<FrozenPhaseInterface>;
+  hotPhaseRolloverEnabled: boolean;
+}
+export class FrozenPhase extends PureComponent<Props> {
+  render() {
+    const {
+      setPhaseData,
+      phaseData,
+      errors,
+      isShowingErrors,
+      hotPhaseRolloverEnabled,
+    } = this.props;
+
+    return (
+      <div id="frozenPhaseContent" aria-live="polite" role="region">
+        <EuiDescribedFormGroup
+          title={
+            <div>
+              <h2 className="eui-displayInlineBlock eui-alignMiddle">
+                <FormattedMessage
+                  id="xpack.indexLifecycleMgmt.editPolicy.frozenPhase.frozenPhaseLabel"
+                  defaultMessage="Frozen phase"
+                />
+              </h2>{' '}
+              {phaseData.phaseEnabled && !isShowingErrors ? <ActiveBadge /> : null}
+              <PhaseErrorMessage isShowingErrors={isShowingErrors} />
+            </div>
+          }
+          titleSize="s"
+          description={
+            <Fragment>
+              <p>
+                <FormattedMessage
+                  id="xpack.indexLifecycleMgmt.editPolicy.frozenPhase.frozenPhaseDescriptionText"
+                  defaultMessage="You are querying your index very infrequently, so you can allocate shards
+                  on the least performant hardware.
+                  Because your queries are slower, you can reduce the number of replicas."
+                />
+              </p>
+              <EuiSwitch
+                data-test-subj="enablePhaseSwitch-frozen"
+                label={
+                  <FormattedMessage
+                    id="xpack.indexLifecycleMgmt.editPolicy.frozenPhase.activateWarmPhaseSwitchLabel"
+                    defaultMessage="Activate frozen phase"
+                  />
+                }
+                id={`${frozenProperty}-${phaseProperty('phaseEnabled')}`}
+                checked={phaseData.phaseEnabled}
+                onChange={(e) => {
+                  setPhaseData(phaseProperty('phaseEnabled'), e.target.checked);
+                }}
+                aria-controls="frozenPhaseContent"
+              />
+            </Fragment>
+          }
+          fullWidth
+        >
+          <Fragment>
+            {phaseData.phaseEnabled ? (
+              <Fragment>
+                <MinAgeInput<FrozenPhaseInterface>
+                  errors={errors}
+                  phaseData={phaseData}
+                  phase={frozenProperty}
+                  isShowingErrors={isShowingErrors}
+                  setPhaseData={setPhaseData}
+                  rolloverEnabled={hotPhaseRolloverEnabled}
+                />
+                <EuiSpacer />
+
+                <NodeAllocation<FrozenPhaseInterface>
```

Comment:
Similar here, the props generics don't include frozen phase, but  type check is not showing it.